### PR TITLE
add review status to application list (to the view only)

### PR DESCRIPTION
**Pull request by andreievg**
Fixes ? (not sure which issue it fixes

needs backend branch:  https://github.com/openmsupply/application-manager-server/pull/227

## Changes

Added filter reviewerId = userId to all application list queries 
Added new fields form view:

```
        numberOfAssignedNotStartedReviews
        numberOfAssignedReviews
        numberOfChangesRequestedReviews
        numberOfDraftReviews
        numberOfReviewsSelfAssignedBySomeoneElse
        numberOfSelfAssignableReviews
        numberOfSubmittedReviews
```

### CODE REVIEW

**Comment by nmadruga** [comment]
I think this would cover #227 partially - but might be good to break down this issue to have more granular actionable displayed after we get each specific parts of the review in place on the ReviewStructure

File: src/utils/graphql/queries/getApplicationsList.query.ts
Comment (nmadruga) [inline src/utils/graphql/queries/getApplicationsList.query.ts:35]: I think this field name could be simplified with something as `numberOfAssignedToAnother`

**Review by nmadruga — APPROVED** [review]
Cool. Simple PR, I think it's ok, just make one suggestion (which involves change to the back-end). 

I haven't seen the `actionable` displayed in the list, I think that would be coming in a future PR (to complete what is required for #227). And also we need some solution to the problem - when review lvl > 1 needs to see how many reviews (from other users) have been submitted/re-submitted so they can take action to start a new consolidation.

**Comment by CarlosNZ** [comment]
I'm just having a look at this now. I think I have a suggestion, but not sure if it'll work, so give me a few minutes before you merge?

File: src/utils/helpers/list/buildQueryFilters.ts
Comment (CarlosNZ) [inline src/utils/helpers/list/buildQueryFilters.ts:19]: This is just a error-protection because graphQL can't accept empty filter objects. It was meant to return everything if no filter object was supplied, but with `isNull: true` it'll return nothing. Since the next line adds a reviewerId to the filter object, we probably don't need this line (maybe comment it out for now?)

**Review by CarlosNZ — COMMENTED** [review]
Just wondering -- why have the application_list view return a massive join that gives a full set of _all_ applications for _every_ reviewer? Wouldn't it be cleaner to just have a second query that fetches the review counts separately? Or, if the review counts are intended to be used for display in particular columns, perhaps those cell components can have their own internal query to fetch this info?

Not sure, just interested why you think this is the best way?

**Comment by andreievg** [comment]
> Just wondering -- why have the application_list view return a massive join that gives a full set of all applications for every reviewer? Wouldn't it be cleaner to just have a second query that fetches the review counts separately? Or, if the review counts are intended to be used for display in particular columns, perhaps those cell components can have their own internal query to fetch this info?

How do you filter, pagenate etc with two queries ?

> Not sure, just interested why you think this is the best way?

As per above, so that we can still filter, pagenate and show counted filters in dashboard

I am also a bit worried about the big join, but i think this solutions should be way better: https://github.com/openmsupply/application-manager-server/pull/227#issuecomment-788670002

**Comment by andreievg** [comment]
Please don't merge

**Comment by andreievg** [comment]
comments will be transferred to this pr: https://github.com/openmsupply/application-manager-server/pull/266

new pr created for this : https://github.com/openmsupply/application-manager-web-app/pull/469

**Pull request closed without merging**
